### ISSUE TITLE
Signed binaries

### ISSUE THREAD

**Issue by Neverbirth** [lix-pm/lix.client#66]
Due to how security works on my company we have problems being able to switch between Haxe versions. I checked with them and it looks like applications should be signed.

It seems also defining the binary metadata like Product name and Copyright could be helpful, but I wouldn't bet my life on it, as I saw in their system the word "Publisher" and that's just a certificate thing.

**Comment by Neverbirth** [comment]
I have dedicated a bit of time into the binary metadata topic. I created a VERSIONINFO resource file (attached), and compiled it. For compiling there are multiple options out there, but I used [GoRC](http://www.godevtool.com/) in the end.

For linking the compiled resource file and the binary there are also several options out there, I also found some tools that allow for adding the resource file after generating the .exe file, but I didn't like too much the idea of using any of those. Sadly TinyCC doesn't support linking the files as far as I could tell. I tried some other portable C compiler called [Digital Mars C Compiler](https://digitalmars.com/download/freecompiler.html), and it worked fully fine, the final file is 27kb IIRC, which is slightly bigger than current one but I think the extra size is minimal, I looked into how to reduce it, but didn't have any luck.

Using TinyCC is possible, but requires some tricks explained [here](http://lists.gnu.org/archive/html/tinycc-devel/2006-10/msg00047.html). It is working fine too.

I didn't get to try out yet running the PostInstall script, but will do tomorrow.

It would be good to come up with some final values for the values of the attributes, and during next week I will discuss with the people of the company if this is enough or signing the application is also needed (in which case it will be needed to change how this exify thing works).

[versionedexify.zip](https://github.com/lix-pm/lix.client/files/2848411/versionedexify.zip)
